apps/HelloAndroidCamera2/jni/HalideFilters.cpp

```diff
@@ -0,0 +1,169 @@
+#include <jni.h>
+#include <android/log.h>
+#include <android/bitmap.h>
+#include <android/native_window_jni.h>
+
+#include <algorithm>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+#define LOGD(...) __android_log_print(ANDROID_LOG_DEBUG, "native", __VA_ARGS__)
+#define LOGE(...) __android_log_print(ANDROID_LOG_ERROR, "native", __VA_ARGS__)
+
+#include "AndroidBufferUtilities.h"
+#include "BufferTFunctions.h"
+#include "YuvFunctions.h"
+#include "deinterleave.h"
+#include "edge_detect.h"
+#include "HalideRuntime.h"
+
+#define DEBUG 1
+
+// Extern functions from the Halide runtime that are not exposed in
+// HalideRuntime.h.
+extern "C" int halide_host_cpu_count();
```

Comment:
Not every OS uses host_cpu_count, so it's not always available, so it's not exposed in HalideRuntime.h. Android does have it though. We could make it part of the API and add a windows implementation (getenv("NUMBER_OF_PROCESSORS")).

halide_current_time_ns could probably go in HalideRuntime.h too